syft/cataloging/archive_search.go

```diff
@@ -0,0 +1,23 @@
+package cataloging
+
+type ArchiveSearchConfig struct {
+	IncludeIndexedArchives   bool `yaml:"include-indexed-archives" json:"include-indexed-archives" mapstructure:"include-indexed-archives"`
+	IncludeUnindexedArchives bool `yaml:"include-unindexed-archives" json:"include-unindexed-archives" mapstructure:"include-unindexed-archives"`
+}
+
+func DefaultArchiveSearchConfig() ArchiveSearchConfig {
+	return ArchiveSearchConfig{
+		IncludeIndexedArchives:   true,
+		IncludeUnindexedArchives: false,
```

Comment:
an example of an indexed archive is zip, since there is a table at the end of the file that can be used to quickly search for matching paths that are contained within the zip without needing to decompress anything. A tar.gz is an example of an unindexed archive, where we need to decompress the contents and seek through the entire file (over each tar header) to understand what paths are contained within the archive. 

This distinction in the configuration allows for "speedy" vs "deep" archive searches. This is not supported by most catalogers yet.